src/read.c

```diff
@@ -1613,7 +1613,10 @@ static avifResult avifDecoderFindMetadata(avifDecoder * decoder, avifMeta * meta
             {
                 uint32_t exifTiffHeaderOffset;
                 AVIF_CHECKERR(avifROStreamReadU32(&exifBoxStream, &exifTiffHeaderOffset),
-                              AVIF_RESULT_BMFF_PARSE_FAILED); // unsigned int(32) exif_tiff_header_offset;
+                              AVIF_RESULT_INVALID_EXIF_PAYLOAD); // unsigned int(32) exif_tiff_header_offset;
+                size_t expectedExifTiffHeaderOffset;
+                AVIF_CHECKRES(avifGetExifTiffHeaderOffset(exifContents.data + 4, exifContents.size - 4, &expectedExifTiffHeaderOffset));
```

Comment:
Optional: To help code analysis, we should use the same data and size values in this line and in line 1622:

```
AVIF_CHECKRES(avifRWDataSet(&image->exif, avifROStreamCurrent(&exifBoxStream), avifROStreamRemainingBytes(&exifBoxStream)));
```